lib/handshake_client.ml

```diff
@@ -26,11 +26,15 @@ let default_client_hello config =
     match version with
     | TLS_1_0 | TLS_1_1 -> List.filter (o not Ciphersuite.ciphersuite_tls12_only) cs
     | TLS_1_2           -> cs
+  and sessionid =
+    match config.session_cache (Cstruct.create 0) with
```

Comment:
This `Cstruct.create 0` is a bit of an odd interface.  Is it mutated or realloced or something by the `session_cache` function?